### PR TITLE
The onDocumentCleaned event is raised wrongly after document.write is called for a document that is still being loaded (fix #253)

### LINKED ISSUE #253: The onDocumentCleaned event is raised wrongly after document.write is called for a document that is still being loaded
Multiple calls to  `document.write`  leads to the page hanging. Because each time`onDocumentCleaned` is raised Hammerhead gets initialized.

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @inikulin @churkin

### CODE REVIEW

File: src/client/sandbox/node/document.js [hunk 1]
```diff
@@ -72,11 +76,11 @@ export default class DocumentSandbox extends SandboxBase {
             return null;
         }
 
-        var isUninitializedIframe = this._isUninitializedIframeWithoutSrc(this.window);
+        var needRiseDocumentCleaned = !this._isUninitializedIframeWithoutSrc() && this._isDocumentBeCleanedAfterWrite();
```
Comment (inikulin) [inline src/client/sandbox/node/document.js]: `needRiseDocumentCleaned` -> `shouldEmitDocumentCleanedEvent`

Comment (inikulin) [inline src/client/sandbox/node/document.js]: `_isDocumentBeCleanedAfterWrite` -> `_isDocumentCleanedAfterWrite`

Comment (inikulin) [inline src/client/sandbox/node/document.js]: Actually you don't need method here, using variable will be just fine:
```js
var documentBeCleanedAfterWrite = this.document.readyState !== 'loading';
```

File: src/client/sandbox/node/document.js [hunk 2]
```diff
@@ -16,16 +16,20 @@ export default class DocumentSandbox extends SandboxBase {
...
+            return this.window !== this.window.top && isIframeWithoutSrc(this.window.frameElement) &&
+                   !IframeSandbox.isIframeInitialized(this.window.frameElement);
         }
         catch (e) {
             return false;
         }
     }
 
+    _isDocumentBeCleanedAfterWrite () {
+        return this.document.readyState !== 'loading';
```
Comment (inikulin) [inline src/client/sandbox/node/document.js]: Why this state indicates that document was cleaned?

**Comment by inikulin** [comment]
\r-

File: src/client/sandbox/node/document.js (hunk 2 above)
Comment (LavrovArtem) [inline src/client/sandbox/node/document.js]: this state indicates that document will be cleaned after ```document.write```, because page alredy loaded

Comment (inikulin) [inline src/client/sandbox/node/document.js]: Well, then it's declared as functions outside `_beforeDocumentCleaned` it doesn't makes much sense. Yet another evidence that it should be used as variable as I suggested in the comment below.

**Comment by LavrovArtem** [comment]
FPR

**Comment by inikulin** [comment]
lgtm

**Comment by LavrovArtem** [comment]
@churkin for this condition
```javascript
(this.readyStateForIE || this.document.readyState) !== 'loading'
```
ie9/ie10 incorrect using only ```loading``` status, other status may be taken from ```this.document.readyState```

**Comment by LavrovArtem** [comment]
FPR

**Comment by inikulin** [comment]
lgtm

**Comment by churkin** [comment]
lgtm

**Pull request merged**